packages/main/src/RPA/Excel/Files.py

```diff
@@ -1206,17 +1216,41 @@ def append_worksheet(self, name=None, content=None, header=False, start=None):
         if header and is_empty:
             sheet.append(columns)
 
-        for row in content:
-            values = [""] * len(columns)
-            for column, value in row.items():
-                try:
-                    index = columns.index(column)
-                    values[index] = value
-                except ValueError:
-                    pass
-            sheet.append(values)
+        if formatting_as_empty:
+            self._append_on_first_empty_based_on_values(content, sheet)
+        else:
+            for row in content:
+                values = [""] * len(columns)
+                for column, value in row.items():
+                    try:
+                        index = columns.index(column)
+                        values[index] = value
+                    except ValueError:
+                        pass
+                sheet.append(values)
+
+        self.active = sheet_name
+
+    def _append_on_first_empty_based_on_values(self, content, sheet):
+        first_empty = None
+        for row in sheet.rows:
+            values = []
+            for cell in row:
+                values.append(cell.value)
+            if all(value is None for value in values):
+                if not first_empty:
+                    first_empty = row[0].row
+            else:
+                first_empty = None
 
-        self.active = name
+        if first_empty:
```

Comment:
⚠️ What if there's no `first_empty` found at all? By judging the overall logic, it looks like we won't append the row at all. (if the last row in the `sheet` contains values)

Reply:
Fixed this one also.